agent/cache/cache.go

```diff
@@ -499,7 +505,13 @@ func (c *Cache) fetch(key string, r getOptions, allowNew bool, attempt uint, ign
 				Index: entry.Index,
 			}
 		}
-
+		if err := entry.RateLimiter.Wait(context.Background()); err != nil {
+			// This should might happen only if:
+			// burstSize is 0, Context is canceled, or the expected wait time exceeds the Context's Deadline.
+			// so, it should not happen, fix this if you plan using something different than:
+			// context.Background()
```

Comment:
Just to be extra sure we don't miss it, I think a comment above the line about the context would be good

```suggestion
		// TODO(context): handle the error without a panic when context is changed to one that is cancellable
		if err := entry.RateLimiter.Wait(context.Background()); err != nil {
			// an error here is a programming error (bursSize is 0) so it is ok to panic, until the context
			// is changed to one that can be cancelled
```